### PR TITLE
TF-831: Add a new recipient in forwarding

### PR DESCRIPTION
**Pull request by ManhNTX**
https://user-images.githubusercontent.com/107173849/185751297-75b0e651-8e0b-47f7-9f7d-fae0f2e1cdbd.mov

https://user-images.githubusercontent.com/107173849/186039878-8e0489e0-8dfc-4d60-a125-a2a6623a29e7.mov

### CODE REVIEW

File: lib/features/manage_account/domain/usecases/add_recipients_in_forwarding_interactor.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+class AddRecipientsInForwardingInteractor {
+  final ManageAccountRepository manageAccountRepository;
+
+  AddRecipientsInForwardingInteractor(this.manageAccountRepository);
+
+  Stream<Either<Failure, Success>> execute(
+      AccountId accountId,
+      AddRecipientInForwardingRequest addRequest,
+      ) async* {
```
Comment (hoangdat) [inline lib/features/manage_account/domain/usecases/add_recipients_in_forwarding_interactor.dart:19]:
```
  Stream<Either<Failure, Success>> execute(
      AccountId accountId,
      AddRecipientInForwardingRequest addRequest,
  ) async* {
```

File: lib/features/manage_account/presentation/model/account_menu_item.dart [hunk 2]
```diff
@@ -9,6 +9,9 @@ enum AccountMenuItem {
   emailRules,
   forward,
   vacation;
+}
+
+extension AccountMenuItemExtension on AccountMenuItem {
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/model/account_menu_item.dart:14]: why we need to do that?

File: lib/features/manage_account/presentation/forward/widgets/forward_header_widget.dart [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+              fontSize: 17,
+              color: Colors.white,
+              fontWeight: FontWeight.w500,
+            ))
+            ..onPressActionClick(() => addEmailForward.call())
+            ..text(
+              AppLocalizations.of(context).addRecipients,
+              isVertical: false,
+            ))
+          .build();
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/forward/widgets/forward_header_widget.dart:65]: all lines should be less indent

File: lib/features/manage_account/presentation/forward/widgets/forward_header_widget.dart [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
```
Comment (hoangdat) [inline lib/features/manage_account/presentation/forward/widgets/forward_header_widget.dart:87]: idem

File: lib/features/emails_forward_creator/presentation/emails_forward_creator_controller.dart [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  late Session? _session;
+  late GetAutoCompleteWithDeviceContactInteractor _getAutoCompleteWithDeviceContactInteractor;
+  late GetAutoCompleteInteractor _getAutoCompleteInteractor;
+
+  final listEmailForwards = RxSet<EmailAddress>();
+
+  EmailsForwardCreatorController(
+      this._getAutoCompleteWithDeviceContactInteractor,
+      this._getAutoCompleteInteractor,
+      );
```
Comment (dab246) [inline lib/features/emails_forward_creator/presentation/emails_forward_creator_controller.dart:36]: remove indent

File: lib/features/emails_forward_creator/presentation/widgets/emails_forward_input_with_drop_list_field_builder.dart [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+import 'package:flutter/material.dart';
+import 'package:flutter_typeahead/flutter_typeahead.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_address.dart';
+import 'package:tmail_ui_user/features/emails_forward_creator/presentation/widgets/emails_forward_input_decoration_builder.dart';
+
+typedef OnSelectedSuggestionAction = Function(EmailAddress emailAddress);
+typedef OnSuggestionCallbackAction = Function(String? pattern);
+typedef OnSummitedCallbackAction = Function(String? pattern);
+
+
```
Comment (dab246) [inline lib/features/emails_forward_creator/presentation/widgets/emails_forward_input_with_drop_list_field_builder.dart:12]: excess line

**Review by dab246 — COMMENTED** [review]
IMO, the `AddRecipient` view should be shaded fullscreen on the web. What do you think? @@ManhNTX  @hoangdat 
Screen Shot 2022-08-22 at 10 44 39

**Comment by ManhNTX** [comment]
> IMO, the `AddRecipient` view should be shaded fullscreen on the web. What do you think? @@ManhNTX @hoangdat Screen Shot 2022-08-22 at 10 44 39

@dab246 IMO, I think it is like dialog of forwarding so it should located in the view of forwarding.

**Comment by hoangdat** [comment]
> IMO, the AddRecipient view should be shaded fullscreen on the web. What do you think? @@ManhNTX @hoangdat

Agree. So weird when not shade menu

**Review by dab246 — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**